### PR TITLE
Adding custom test debug info API

### PR DESCRIPTION
**Pull request by scalvert**
This PR is a WIP of an API allowing applications and addons to add their own custom test debug info. Providing this API can allow for tailored information that can assist in debugging test issues.

### CODE REVIEW

**Comment by scalvert** [comment]
@rwjblue any thoughts on this?

**Comment by scalvert** [comment]
Done, and also simplified the interface. The conditional check can now be done inside the `DebugHelper.toConsole` method.

**Comment by scalvert** [comment]
@rwjblue I think this is good to go once CI passes.

File: addon-test-support/@ember/test-helpers/-internal/debug-info-helpers.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  name: string;
+  log: () => void;
+}
+
+export const debugInfoHelpers = new Set<DebugInfoHelper>();
+
+/**
+ * Registers a custom debug info helper to augment the output for test isolation validation.
+ *
+ * @public
```
Comment (rwjblue) [inline addon-test-support/@ember/test-helpers/-internal/debug-info-helpers.ts]: 😱 :rage4: `*`

Comment (scalvert) [reply addon-test-support/@ember/test-helpers/-internal/debug-info-helpers.ts]: Sorry, what's wrong here? I'm confused :)

Comment (rwjblue) [reply addon-test-support/@ember/test-helpers/-internal/debug-info-helpers.ts]: OH YOU KNOW WHAT YOU DID

Comment (scalvert) [reply addon-test-support/@ember/test-helpers/-internal/debug-info-helpers.ts]: Hehe

**Comment by scalvert** [comment]
I'm seriously not trolling you (tempting though it is!), but there's a bunch of other modules in here that use the `*` block comment format that you despise so much. I'm happy to follow whatever the standard should be. Let me know.

**Pull request merged**